tests/conftest.py

```diff
@@ -459,6 +460,13 @@ def deactive_venv():
     # reset sys.meta_path because it might contain finders for editable installs, make sure to keep the same object
     sys.meta_path.clear()
     sys.meta_path.extend(old_meta_path)
+    sys.path_hooks.clear()
+    sys.path_hooks.extend(old_path_hooks)
+    # Clear cache for sys.path_hooks
+    sys.path_importer_cache.clear()
+    sys.modules.clear()
+    sys.modules.update(old_sys_modules)
```

Comment:
I've considered this in the past but I decided against it because it requires reloading everything, even submodules of the `inmanta-core` package, test modules, third party libraries, ... I wonder it wouldn't be better to be a bit more directed here and just keep the `unload_inmanta_plugins` and `unload_modules_for_path` calls (but consolidated and improved as you did). Wdyt?

Reply:
Fine by me.